python/mxnet/gluon/contrib/rnn/conv_rnn_cell.py

```diff
@@ -0,0 +1,971 @@
+        return ('',)
+
+    def hybrid_forward(self, F, inputs, states, i2h_weight,
+                       h2h_weight, i2h_bias, h2h_bias):
+        prefix = 't%d_'%self._counter
+        i2h, h2h = self._conv_forward(F, inputs, states,
+                                      i2h_weight, h2h_weight, i2h_bias, h2h_bias,
+                                      prefix)
+        output = self._get_activation(F, i2h + h2h, self._activation,
+                                      name=prefix+'out')
+        return output, [output]
+
+
+class Conv1DRNNCell(_ConvRNNCell):
+    r"""1D Convolutional RNN cell.
+
+    .. math::
+
+        h_t = tanh(W_i \ast x_t + R_i \ast h_{t-1} + b_i)
+
+    Parameters
+    ----------
+    hidden_channels : int
+        Number of output channels.
+    input_shape : tuple of int
+        Input tensor shape at each time step for each sample, excluding dimension of the batch size
+        and sequence length. Must be consistent with `conv_layout`.
+        For example, for layout 'NCW' the shape should be (C, W).
+    activation : str or Block, default 'tanh'
+        Type of activation function.
+        If argument type is string, it's equivalent to nn.Activation(act_type=str). See
+        :func:`~mxnet.ndarray.Activation` for available choices.
+        Alternatively, other activation blocks such as nn.LeakyReLU can be used.
+    i2h_kernel : int or tuple of int, default (3,)
+        Input convolution kernel sizes.
+    i2h_stride : int or tuple of int, default (1,)
+        Input convolution stride sizes.
+    i2h_pad : int or tuple of int, default (1,)
+        Pad for input convolution.
+    i2h_dilate : int or tuple of int, default (1,)
+        Input convolution dilate.
+    h2h_kernel : int or tuple of int, default (3,)
+        Recurrent convolution kernel sizes. Only odd-numbered sizes are supported.
+    h2h_dilate : int or tuple of int, default (1,)
+        Recurrent convolution dilate.
+    i2h_weight_initializer : str or Initializer
+        Initializer for the input weights matrix, used for the input convolutions.
+    h2h_weight_initializer : str or Initializer
+        Initializer for the recurrent weights matrix, used for the input convolutions.
+    i2h_bias_initializer : str or Initializer, default zeros
+        Initializer for the input convolution bias vectors.
+    h2h_bias_initializer : str or Initializer, default zeros
+        Initializer for the recurrent convolution bias vectors.
+    conv_layout : str, default 'NCW'
+        Layout for all convolution inputs, outputs and weights. Options are 'NCW' and 'NWC'.
+    prefix : str, default 'conv_rnn_'
+        Prefix for name of layers (and name of weight if params is None).
+    params : RNNParams, default None
+        Container for weight sharing between cells. Created if None.
+    """
+    def __init__(self, hidden_channels, input_shape, activation='tanh',
+                 i2h_kernel=(3,), i2h_stride=(1,), i2h_pad=(1,), i2h_dilate=(1,),
```

Comment:
Why is there a default value for kernel?

Reply:
state what you'd like please

Reply:
https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/rnn/rnn_cell.py#L1185-L1196

Reply:
I don't know. But considering Conv2D doesn't set default value for kernel, what's the reason for adding one here? Also default padding is 0 for Conv2D.

Reply:
@sxjscience what would you recommend?

Reply:
Either way this needs to be better documented.

Reply:
I agree that we should remove the default.

Reply:
I think we should remove the default value of i2h_kernel and h2h_kernel and keep the default value of i2h_pad, i2h_stride,... h2h_dilate to be the same as that of the convolution layer.

Reply:
Will do

Reply:
I have updated the parameters. In addition I made two other changes:
1. i2h and h2h strides are tied. Otherwise i2h and h2h may not match in size
2. activation is moved to last because it has default value of tanh, which can't precede the required parameters.